### PR TITLE
feat(stability): migrate component

### PR DESCRIPTION
**Pull request by AugustinMauroy**
#5309

### CODE REVIEW

File: components/Api/Stability/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import type { PropsWithChildren } from 'react';
+
+type Props = PropsWithChildren<{ stability: number }>;
```
Comment (ovflowd) [inline components/Api/Stability/index.tsx]: Type should be close to the component 👀 

Also...

```suggestion
type StabilityProps = PropsWithChildren<{ stability: number }>;
```

File: components/Api/Stability/index.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    case 2:
+      return styles.stabilityLevel2;
+    case 3:
+      return styles.stabilityLevel3;
+    default:
+      throw new Error(`Unknown stability level: ${stability}`);
+  }
+};
+
+const Stability = ({ stability, children }: Props) => (
```
Comment (ovflowd) [inline components/Api/Stability/index.tsx]:
```suggestion
const Stability: React.FC<StabilityProps> = ({ stability, children }) => (
```

I see myself commenting this same thing on every PR of yours 🙈

**Review by ovflowd — APPROVED** [review]
Left a few comments, but overall, LGTM! (And thanks for the contribution!)

File: components/Api/Stability/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import Stability from './index';
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
+
+type Story = StoryObj<typeof Stability>;
+type Meta = MetaObj<typeof Stability>;
+
+export const Stability0: Story = {
```
Comment (ovflowd) [inline components/Api/Stability/index.stories.tsx]: I feel that only one Story might be enough. And allowing the user to select both the children and the `stability` number from a dropdown. (That can be done through the `select` control (https://storybook.js.org/docs/react/essentials/controls#annotation)

Comment (AugustinMauroy) [reply components/Api/Stability/index.stories.tsx]: I'll take a look for this 😁

File: components/Api/Stability/index.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    case 2:
+      return styles.stabilityLevel2;
+    case 3:
+      return styles.stabilityLevel3;
+    default:
+      throw new Error(`Unknown stability level: ${stability}`);
+  }
+};
+
+const Stability: React.FC<StabilityProps> = ({ stability, children }) => (
```
Comment (ovflowd) [inline components/Api/Stability/index.tsx]: You need to import the FC types

```suggestion
const Stability: FC<StabilityProps> = ({ stability, children }) => (
```

Comment (AugustinMauroy) [reply components/Api/Stability/index.tsx]: yeah sure

File: components/Api/Stability/index.stories.tsx (hunk 3 above)
Comment (AugustinMauroy) [reply components/Api/Stability/index.stories.tsx]: What do you thinks about my change ?

File: components/Api/Stability/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import Stability from './index';
+import type { Meta as MetaObj, StoryObj } from '@storybook/react';
+
+type Story = StoryObj<typeof Stability>;
+type Meta = MetaObj<typeof Stability>;
+
+export const Default: Story = {
+  args: {
+    stability: 0,
+    children: 'children',
```
Comment (ovflowd) [inline components/Api/Stability/index.stories.tsx]: You could add some example test here instead of just "children" which is confusing

Comment (AugustinMauroy) [reply components/Api/Stability/index.stories.tsx]: Yeah!

Comment (AugustinMauroy) [reply components/Api/Stability/index.stories.tsx]: I don't found any thing to control children in same time of stability

Comment (ovflowd) [reply components/Api/Stability/index.stories.tsx]: What? I think you misunderstood me. I'm just asking you to change the value of "children" to something more meangiful. For example:

```suggestion
    children: 'Insert the text of your Alert here',
```

Comment (AugustinMauroy) [reply components/Api/Stability/index.stories.tsx]: I mean : it's possible to change children in same time of stability?

**Comment by ovflowd** [comment]
**This PR got accidentally closed because the branch got deleted. Please feel free to reopen the PR.**

**Comment by AugustinMauroy** [comment]
Close in favor of proper branch

**Comment by ovflowd** [comment]
wdym @AugustinMauroy?

**Comment by AugustinMauroy** [comment]
the commit history of this pr is different that "new" `major/website-redesign`

**Comment by ovflowd** [comment]
Well you can just checkout the branch and cherry pick and then force push. No need to close the PR...

File: components/Api/Stability/__test__/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import { render } from '@testing-library/react';
+import { IntlProvider } from 'react-intl';
+import Stability from '..';
+
+describe('Stability', () => {
```
Comment (ovflowd) [inline components/Api/Stability/__test__/index.test.tsx]: Can you remove these unit tests? We now use snapshot testing provided by Storybook, @Harkunwar could you write here if any other step needs to be taken 👀

**Review by ovflowd — COMMENTED** [review]
Mostly LGTM! A little comment about the unit tests s

File: components/Api/Stability/__test__/index.test.tsx (hunk 6 above)
Comment (AugustinMauroy) [reply components/Api/Stability/__test__/index.test.tsx]: Done!

File: components/Api/Stability/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+  a {
+    color: #fff;
+
+    code {
+      color: #fff !important;
+    }
+  }
+
+  &Level0 {
```
Comment (ovflowd) [inline components/Api/Stability/index.module.scss:21]: Can the letter L be lowercase on all these "levels"?

**Review by ovflowd — APPROVED** [review]
Just left a tiny comment

File: components/Api/Stability/index.module.scss (hunk 7 above)
Comment (AugustinMauroy) [reply components/Api/Stability/index.module.scss:21]: i will take a look

Comment (AugustinMauroy) [reply components/Api/Stability/index.module.scss:21]: Nop we put on upper case to write that:
`styles.stabilityLevel0`

File: components/Api/Stability/index.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    case 1:
+      return styles.stabilityLevel1;
+    case 2:
+      return styles.stabilityLevel2;
+    case 3:
+      return styles.stabilityLevel3;
+    default:
+      throw new Error(`Unknown stability level: ${stability}`);
+  }
+};
```
Comment (MrJithil) [inline components/Api/Stability/index.tsx]: Can we optimise this function?


I think, its better to reduce the code as below:

```suggestion
 const getStabilityClass = (stability: number) => {
    const style = styles[`stabilityLevel${stability}`]
    if(!style)
        throw new Error(`Unknown stability level: ${stability}`);
 };
```

File: components/Api/Stability/index.tsx [hunk 9]
```diff
@@ -0,0 +1,23 @@
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import type { PropsWithChildren, FC } from 'react';
+
+type StabilityProps = PropsWithChildren<{ stability: number }>;
+
+ const getStabilityClass = (stability: number) => {
+    const style = styles[`stabilityLevel${stability}`]
+    if(!style)
```
Comment (HinataKah0) [inline components/Api/Stability/index.tsx]: Please run lint, thanks!
```js
if (!style) {
  throw new Error(`Unknown stability level: ${stability}`);
}
```

Comment (AugustinMauroy) [reply components/Api/Stability/index.tsx]: done !

File: components/Api/Stability/index.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import type { PropsWithChildren, FC } from 'react';
+
+type StabilityProps = PropsWithChildren<{ stability: number }>;
+
+const getStabilityClass = (stability: number) => {
+  const style = styles[`stabilityLevel${stability}`];
+  if (!style) throw new Error(`Unknown stability level: ${stability}`);
+};
```
Comment (MrJithil) [inline components/Api/Stability/index.tsx]: Don't we need a `return style` at the end of the function?
```suggestion
const getStabilityClass = (stability: number) => {
  const style = styles[`stabilityLevel${stability}`];
  if (!style) throw new Error(`Unknown stability level: ${stability}`);
  return style;
};
```

**Review by HinataKah0 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**